tsl/src/continuous_aggs/invalidation.c

```diff
@@ -313,27 +313,43 @@ invalidation_hyper_log_add_entry(int32 hyper_id, int64 start, int64 end)
  * hypertable.
  */
 void
-invalidation_add_entry(const Hypertable *ht, ContinuousAggHypertableStatus caggstatus,
-					   int32 entry_id, int64 start, int64 end)
+continuous_agg_invalidate(const Hypertable *raw_ht, int64 start, int64 end)
 {
-	Assert(ht != NULL);
+	continuous_agg_invalidate_with_materialization(raw_ht, NULL, start, end);
+}
+
+void
+continuous_agg_invalidate_with_materialization(const Hypertable *raw_ht, const Hypertable *mat_ht,
+											   int64 start, int64 end)
+{
+	ContinuousAggHypertableStatus caggstatus = ts_continuous_agg_hypertable_status(raw_ht->fd.id);
+
+	Assert(raw_ht != NULL);
 
-	if (hypertable_is_distributed(ht))
+	if (hypertable_is_distributed(raw_ht))
 	{
-		remote_invalidation_log_add_entry(ht, caggstatus, entry_id, start, end);
+		int32 entry_id;
+
+		if (mat_ht)
+		{
+			entry_id = mat_ht->fd.id;
+			caggstatus = HypertableIsMaterialization;
+		}
+		else
+			entry_id = raw_ht->fd.id;
+
+		remote_invalidation_log_add_entry(raw_ht, caggstatus, entry_id, start, end);
 	}
 	else
 	{
-		ContinuousAggHypertableStatus caggstatus = ts_continuous_agg_hypertable_status(ht->fd.id);
-
 		switch (caggstatus)
 		{
 			case HypertableIsMaterialization:
-				invalidation_cagg_log_add_entry(ht->fd.id, start, end);
+				invalidation_cagg_log_add_entry(raw_ht->fd.id, start, end);
```

Comment:
Looks like this needs to be covered via tests.

Reply:
This line is wrong because it should add an entry of `mat_ht`